### PR TITLE
SCC 3908 - Clean up redirect-targets script

### PR DESCRIPTION
**Pull request by nonword**
Some light cleanup and documentation to support testing deployments against our [newly cleaned up redirect targets sheet](https://docs.google.com/spreadsheets/d/1055Y98c_4l-NXWyzoiUhBSkay4SYZMeKEc9-LGhGoqw/edit?gid=1152476292#gid=1152476292)

87 of 94 expectations are passing in prod. Of the 7 failures, most are edge case Legacy paths that we never appear to have handled perfectly. When run against QA, there are a few additional failures due to oclc lookups that succeed in prod but not qa for data reasons.

https://newyorkpubliclibrary.atlassian.net/browse/SCC-3908

### CODE REVIEW

File: test/test-deployment-redirects.js [hunk 1]
```diff
@@ -18,30 +18,45 @@ const chalk = require('chalk')
 
 const argv = require('minimist')(process.argv.slice(2))
 
-const input = argv.input || './encore-vega-redirects.csv'
+const input = argv.input || './redirect-expectations.csv'
+
+const changeToQa = (url) => {
+  return url
+    // Prod Encore to Test Encore:
+    // Note this has another valid hostname: https://nypl-encore-test.
```
Comment (charmingduchess) [inline test/test-deployment-redirects.js:26]: Are we maintaining coverage for `borrow.nypl.org` still?

File: test/test-deployment-redirects.js [hunk 2]
```diff
@@ -56,11 +71,15 @@ const testRow = (row) => {
...
       const matched = redirect === row.target
       if (matched) {
-        console.log(chalk.green('  Matched'))
+        console.log(chalk.green(`  Matched: ${redirect}`))
+
       } else {
-        console.error(chalk.red('  Failed to match:'))
-        console.error(chalk.red(`    Expected redirect: ${row.target}`))
-        console.error(chalk.red(`    Actual redirect:   ${redirect}`))
+        let partialMatch = redirect.indexOf(row.target) === 0 ? 'starts-with' : false
```
Comment (charmingduchess) [inline test/test-deployment-redirects.js:77]: what case would this match on?

File: test/test-deployment-redirects.js (hunk 1 above)
Comment (nonword) [reply test/test-deployment-redirects.js:26]: This app doesn't handle traffic on `borrow.nypl.org`, but does issue redirects to that domain. Do you mean `browse.nypl.org`?

File: test/test-deployment-redirects.js (hunk 2 above)
Comment (nonword) [reply test/test-deployment-redirects.js:80]: Added inline comment. Comes up a lot where the expected URL contains an extra query param that is not present on the actual redirect.

File: test/test-deployment-redirects.js (hunk 1 above)
Comment (charmingduchess) [reply test/test-deployment-redirects.js:26]: oh yes i mixed up the browse and borrow. borrow is the new domain but that's being handled w reverse proxy biz?

**Review by charmingduchess — APPROVED** [review]
🥳

File: test/test-deployment-redirects.js (hunk 1 above)
Comment (nonword) [reply test/test-deployment-redirects.js:26]: Yes `borrow` is the new domain. I believe it's just a CNAME pointing `borrow.nypl` to the Innovative servers.

**Pull request merged**